### DISCUSSION TITLE
How can I make a custom button with themes?

### DISCUSSION THREAD

**Discussion by sallar** [tamagui/tamagui#1084] [General]
Hi. 

I want to have a custom Button that can be themed.
I'm doing this:

```tsx
export const SystemButton = styled(Button, {
  name: 'SystemButton',
  fontWeight: '700',
  backgroundColor: '$background',
})

export type SystemButtonProps = GetProps<typeof SystemButton>
```

And then in themes.tsx:

```tsx
export const themes = addChildren(
  {
    light,
    dark,
  },
  (name, theme) => ({
    SystemButton: {
      ...theme,
      background: tokens.color.black
    }
  })
)
```

But the button component doesn't get any of the defined colors in my theme. 

Am I doing something wrong?

**Comment by trymbill** [comment]
I'm experiencing the same thing. My `styled` button is happy to get a `backgroundColor: '$primary"` but if I give the `<Button />` component a `theme="primary"` it doesn't pick up the theme's `$background` token.

---

Ah never mind, you have to style `ButtonFrame` and `ButtonText` and combine those into a `Button`, just like they do here: https://github.com/tamagui/tamagui/blob/master/apps/kitchen-sink/src/usecases/StyledButtonTheme.tsx
